File: modelo/horasmateria.cs
using System.Data;
using System;
public class horasmateria : oObjeto
{
    public int numsemana { get; set; } // del 0{domingo} al 6{sábado}
    public int Idmateria {get;set;}
    public int Idhoradia { get; set; }  

    public horasmateria() { }
    public horasmateria(DataRow dr)
    {
        this.Id = Convert.ToInt32(dr["id"]);
        this.numsemana = Convert.ToInt32(dr["numsemana"]);
        this.Idmateria = Convert.ToInt32(dr["idmateria"]);
        this.Idhoradia = Convert.ToInt32(dr["idhoradia"]);
    }

}

File: ClientApp/src/app/clases/horasmateria.ts
export class horasmateria
{
    id: number; 
    idmateria: number;
    idhoradia: number;
    numsemana: number;
     
    
    constructor(obj:any)
    {
        this.id= +obj.id;
        this.idmateria = +obj.idmateria;
        this.idhoradia= +obj.idhoradia;
        this.numsemana= +obj.numsemana; 
    }
}